packages/api/lib/granules.js

```diff
@@ -63,10 +83,150 @@ function getGranuleProductVolume(granuleFiles = []) {
+      granule_id: apiGranule.granuleId,
+      collection_cumulus_id: await collectionPgModel.getRecordCumulusId(
+        dbClient,
+        { name, version }
+      ),
+    });
+  } catch (error) {
+    // If the granule or associated record hasn't been migrated yet
+    // run the 'original' dynamo update
+    if (error.name !== 'RecordDoesNotExist') {
+      throw error;
+    }
+    log.info(`Granule ${JSON.stringify(apiGranule)} has not been migrated yet, updating DynamoDb records only`);
+    writeToPostgres = false;
+  }
+
+  const updatedFiles = [];
+  const moveFileParams = generateMoveFileParams(apiGranule.files, destinations);
+  const moveFilePromises = moveFileParams.map(async (moveFileParam) => {
+    const { file } = moveFileParam;
+    try {
+      // Update the datastores, then move files
+      await dbClient.transaction(async (trx) => {
+        const updatedFile = await moveGranuleFile(
+          moveFileParam,
+          filesPgModel,
+          trx,
+          postgresCumulusGranuleId,
+          writeToPostgres
+        );
+        updatedFiles.push(renameProperty('name', 'fileName', updatedFile));
+      });
+      // Add updated file to postgresDatabase
+    } catch (error) {
+      updatedFiles.push(file);
+      log.error(`Failed to move file ${JSON.stringify(file)} -- ${JSON.stringify(error.message)}`);
+      throw error;
+    }
+  });
+
+  const moveResults = await Promise.allSettled(moveFilePromises);
+  await granulesModel.update(
+    { granuleId: apiGranule.granuleId },
+    {
+      files: updatedFiles,
+    }
+  );
+  const moveGranuleErrors = moveResults.filter((r) => r.reason);
+  return { updatedFiles, moveGranuleErrors };
+}
+
+/**
+ * Move a granule's files to destinations specified
+ *
+ * @param {Object} apiGranule - the granule record object
+ * @param {Array<{regex: string, bucket: string, filepath: string}>} destinations
+ *    - list of destinations specified
+ *    regex - regex for matching filepath of file to new destination
+ *    bucket - aws bucket of the destination
+ *    filepath - file path/directory on the bucket for the destination
+ * @param {string} distEndpoint - distribution endpoint URL
+ * @param {Object} granulesModel - An instance of an API Granule granulesModel
+ * @returns {Promise<undefined>} undefined
+ */
+async function moveGranule(apiGranule, destinations, distEndpoint, granulesModel) {
+  log.info(`granules.move ${apiGranule.granuleId}`);
+
+  const bucketsConfig = await s3Utils.getJsonS3Object(
+    process.env.system_bucket,
+    getBucketsConfigKey(process.env.stackName)
+  );
+
+  const bucketTypes = Object.values(bucketsConfig)
+    .reduce(
+      (acc, { name, type }) => ({ ...acc, [name]: type }),
+      {}
+    );
+
+  const distributionBucketMap = await s3Utils.getJsonS3Object(
+    process.env.system_bucket,
+    getDistributionBucketMapKey(process.env.stackName)
+  );
+
+  const {
```

Comment:
The code preceding this is lifted heavily from the granule model move, following it's been updated to use `moveGranuleFilesAndUpdateDatastore` to handle the S3 move/database update, and handle moveGranuleErrors.